sunpy/map/header_helper.py

```diff
@@ -361,3 +361,52 @@ def _set_instrument_meta(meta_wcs, instrument, telescope, observatory, wavelengt
     'CD2_2':
     'Matrix element CDi_j describing the rotation required to align solar North with the top of the image.'
 }
+
+
+def carrington_header(date, observer_coordinate, *, shape_out, projection_code="CAR"):
+    """
+    Construct a FITS-WCS header for a Carrington coordinate frame.
+    The date-time and observer coordinate of the new coordinate frame
+    are taken from the input map. The resulting WCS covers the full surface
+    of the Sun, and has a reference coordinate at (0, 0) degrees Carrington
+    Longitude/Latitude.
+
+    Parameters
+    ----------
+    date :
+        Date for the output header.
+    observer_coordinate :
+        Observer coordinate for the output header.
+    shape_out : [int, int]
+        Output map shape, number of pixels in (latitude, longitude).
+    projection_code : {'CAR', 'CEA'}
+        Projection to use for the latitude coordinate.
+
+    Returns
+    -------
+    `~sunpy.util.MetaDict`
+    """
+    valid_codes = {"CAR", "CEA"}
+    if projection_code not in valid_codes:
+        raise ValueError(f"projection_code must be one of {valid_codes}")
+
+    frame_out = SkyCoord(
+        0 * u.deg,
+        0 * u.deg,
+        frame="heliographic_carrington",
+        obstime=date,
+        observer=observer_coordinate,
+    )
+
+    if projection_code == "CAR":
+        scale = [360 / int(shape_out[0]), 180 / int(shape_out[1])] * u.deg / u.pix
+    elif projection_code == "CEA":
+        # Since, this map uses the cylindrical equal-area (CEA) projection,
+        # the spacing needs to be to 180/pi times the sin(latitude)
+        # spacing
+        # Reference: Section 5.5, Thompson 2006
+        scale = [360 / int(shape_out[0]), 180 / int(shape_out[1]) / (np.pi / 2)] * u.deg / u.pix
```

Comment:
```suggestion
        scale = [360 / int(shape_out[0]), (180 / np.pi) * (2 / int(shape_out[1]))] * u.deg / u.pix
```
If you want to keep things a bit broken out for clarity, you ought to group the elements more logically.  The `180 / np.pi` is the conversion from radians to degrees, and the `2 / int(shape_out[1])` is the spacing of sin(latitude) bins.  Note how my suggested arrangement corresponds with the preceding comment.